### PR TITLE
K8s provisioner changes

### PR DESCRIPTION
**Pull request by jpdhas**
* Removed ansible-vault:  The only value that is in the ansible vault is the konstructor key. If it is passed in as an environment variable, we don't have to update the ansible vault everytime we rotate it
* Enable oidc for the clusters in the provisioner

### CODE REVIEW

**Comment by jpdhas** [comment]
[Update provisioner with API params for OIDC](https://trello.com/c/PfLZiMFy/124-update-provisioner-with-api-params-for-oidc)

File: README.md [hunk 1]
```diff
@@ -51,6 +51,8 @@ Here are the steps for provisioning a new cluster:
         -e "CLUSTER_NAME=$CLUSTER_NAME" \
         -e "CLUSTER_ENVIRONMENT=$CLUSTER_ENVIRONMENT" \
         -e "ENVIRONMENT_TYPE=$ENVIRONMENT_TYPE" \
+        -e "KONSTRUCTOR_API_KEY=$KONSTRUCTOR_API_KEY" \
```
Comment (sbuliarca) [inline README.md:54]: This is a one time key only, right?

We should update also the LP entry with the new params & how to get them.

Comment (sbuliarca) [reply README.md]: On this note, this should not be merged in master until we also update the Jenkins job

File: sh/update.sh [hunk 2]
```diff
@@ -11,4 +11,5 @@ aws_secret_access_key=${AWS_SECRET_ACCESS_KEY} \
 cluster_name=${CLUSTER_NAME} \
 platform=${PLATFORM} \
 cluster_environment=${CLUSTER_ENVIRONMENT} \
+oidc_issuerurl=${OIDC_ISSUER_URL} \
```
Comment (sbuliarca) [inline sh/update.sh]: Please do not merge this until we also update the Jenkins job for updating a cluster

File: sh/provision.sh [hunk 3]
```diff
@@ -9,7 +9,9 @@ aws_region=${AWS_REGION} \
 aws_access_key=${AWS_ACCESS_KEY} \
 aws_secret_access_key=${AWS_SECRET_ACCESS_KEY} \
 cluster_name=${CLUSTER_NAME} \
-share_cluster_credentials=${SHARE_CLUSTER_CREDENTIALS} \
-platform=${PLATFORM} \
+cluster_environment=${CLUSTER_ENVIRONMENT} \
 environment_type=${ENVIRONMENT_TYPE} \
-cluster_environment=${CLUSTER_ENVIRONMENT} "
+konstructor_api_key=${KONSTRUCTOR_API_KEY} \
+oidc_issuerurl=${OIDC_ISSUER_URL} \
```
Comment (tamas-molnar) [inline sh/provision.sh]: Can you please rename `oidc_issuerurl` to `oidc_issuer_url` so it's consistent with the rest of the names?

File: sh/update.sh (hunk 2 above)
Comment (sbuliarca) [reply sh/update.sh]: I managed to update the pipeline library of the job and it is using ATM the version that is able to handle the new parameter.

File: README.md (hunk 1 above)
Comment (jpdhas) [reply README.md:54]: Updated the lastpass entry 👍

File: sh/provision.sh (hunk 3 above)
Comment (jpdhas) [reply sh/provision.sh]: Done 👍

**Review by efinlay24 — APPROVED** [review]

**Pull request closed without merging**